### PR TITLE
Match partial query strings in search results

### PR DESCRIPTION
**Pull request by AndrewVos**
image

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,7 +32,7 @@ source 'https://rubygems.org' do
   gem 'font-awesome-rails'
   gem 'audited'
   gem 'rails-observers', github: 'rails/rails-observers'
-  gem 'pg_search'
+
```
Comment (benlovell) [inline Gemfile]: Remove this.

File: app/models/appointment.rb [hunk 2]
```diff
@@ -54,13 +45,39 @@ def self.copy_or_new_by(id)
...
+    like_query = "%#{query}%"
+
+    select('appointments.*')
+      .ts_headline_select(:appointments, :id, query)
+      .ts_headline_select(:appointments, :first_name, query)
+      .ts_headline_select(:appointments, :last_name, query)
+      .ts_headline_select(:users, :name, query)
+      .where(
+        'appointments.id::text ILIKE :q OR appointments.first_name ILIKE :q OR appointments.last_name ILIKE :q OR users.name ILIKE :q',
+        q: "%#{query}%"
```
Comment (benlovell) [inline app/models/appointment.rb]: This is meant to be `like_query` I think.

File: app/presenters/appointment_search_result_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+class AppointmentSearchResultPresenter < SimpleDelegator
+  DATE_FORMAT = '%d %B %Y %H:%M'.freeze
```
Comment (benlovell) [inline app/presenters/appointment_search_result_presenter.rb]: Is this much different to the usual dates we use elsewhere?

Comment (AndrewVos) [inline app/presenters/appointment_search_result_presenter.rb]: I'm not sure, it's just part of a rename though...

File: db/migrate/20161123155920_install_trigram.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class InstallTrigram < ActiveRecord::Migration[5.0]
+  def self.up
```
Comment (benlovell) [inline db/migrate/20161123155920_install_trigram.rb]: These don't need to be prefixed with `self`.

Comment (AndrewVos) [inline db/migrate/20161123155920_install_trigram.rb]: Sorry didn't see this. Done now

**Comment by benlovell** [comment]
You have a typo in the commit message for 'highlight'.

**Comment by benlovell** [comment]
Squash it down and merge freely. 🚢 🐑

**Comment by benlovell** [comment]
Well played sir. 💯 🛡

**Pull request merged**